src/analyses/mutexEventsAnalysis.ml

```diff
@@ -65,41 +62,27 @@ struct
   let return ctx exp fundec : D.t =
     (* deprecated but still valid SV-COMP convention for atomic block *)
     if get_bool "ana.sv-comp.functions" && String.starts_with fundec.svar.vname "__VERIFIER_atomic_" then
-      ctx.emit (Events.Unlock verifier_atomic)
+      ctx.emit (Events.Unlock (LockDomain.Addr.from_var LF.verifier_atomic_var))
 
   let body ctx f : D.t =
     (* deprecated but still valid SV-COMP convention for atomic block *)
     if get_bool "ana.sv-comp.functions" && String.starts_with f.svar.vname "__VERIFIER_atomic_" then
-      ctx.emit (Events.Lock (verifier_atomic, true))
+      ctx.emit (Events.Lock (LockDomain.Addr.from_var LF.verifier_atomic_var, true))
 
   let special (ctx: (unit, _, _, _) ctx) lv f arglist : D.t =
     let remove_rw x = x in
-    let unlock remove_fn =
-      match f.vname, arglist with
-      | _, [arg]
-      | ("spin_unlock_irqrestore" | "_raw_spin_unlock_irqrestore"), [arg; _] ->
-        List.iter (fun e ->
-            ctx.split () [Events.Unlock (remove_fn e)]
-          ) (eval_exp_addr (Analyses.ask_of_ctx ctx) arg);
-        raise Analyses.Deadcode
-      | _ -> failwith "unlock has multiple arguments"
+    let unlock arg remove_fn =
+      List.iter (fun e ->
+          ctx.split () [Events.Unlock (remove_fn e)]
+        ) (eval_exp_addr (Analyses.ask_of_ctx ctx) arg);
+      raise Analyses.Deadcode
     in
     let desc = LF.find f in
     match desc.special arglist, f.vname with
-    | _, "_lock_kernel" ->
-      ctx.emit (Events.Lock (big_kernel_lock, true))
-    | _, "_unlock_kernel" ->
-      ctx.emit (Events.Unlock big_kernel_lock)
-    | Lock { try_ = failing; write = rw; return_on_success = nonzero_return_when_aquired; _ }, _ ->
-      begin match f.vname, arglist with
-        | _, [arg]
-        | "spin_lock_irqsave", [arg; _] ->
-          (*print_endline @@ "Mutex `Lock "^f.vname;*)
-          lock ctx rw failing nonzero_return_when_aquired (Analyses.ask_of_ctx ctx) lv arg
-        | _ -> failwith "lock has multiple arguments"
-      end
-    | Unlock _, "__raw_read_unlock"
-    | Unlock _, "__raw_write_unlock"  ->
+    | Lock { lock = arg; try_ = failing; write = rw; return_on_success = nonzero_return_when_aquired }, _ ->
+      lock ctx rw failing nonzero_return_when_aquired (Analyses.ask_of_ctx ctx) lv arg
+    | Unlock arg, ("__raw_read_unlock" | "__raw_write_unlock") ->
+      (* TODO: why is this needed? *)
```

Comment:
Why is there some special magic when unlocking raw kernel mutexes, but not when locking?